support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -0,0 +1,268 @@
+	space,
+	textSans,
+	until,
+} from '@guardian/source-foundations';
+import { Column, Columns, Container } from '@guardian/source-react-components';
+import {
+	FooterLinks,
+	FooterWithContents,
+} from '@guardian/source-react-components-development-kitchen';
+import { useEffect, useState } from 'react';
+import {
+	number,
+	object,
+	type Output,
+	parse,
+	picklist,
+	record,
+	string,
+} from 'valibot'; // 1.54 kB
+import { Box, BoxContents } from 'components/checkoutBox/checkoutBox';
+import { CheckoutHeading } from 'components/checkoutHeading/checkoutHeading';
+import { Header } from 'components/headers/simpleHeader/simpleHeader';
+import { ContributionsOrderSummary } from 'components/orderSummary/contributionsOrderSummary';
+import { PageScaffold } from 'components/page/pageScaffold';
+import { SecureTransactionIndicator } from 'components/secureTransactionIndicator/secureTransactionIndicator';
+import type { Currency } from 'helpers/internationalisation/currency';
+import { currencies } from 'helpers/internationalisation/currency';
+import { renderPage } from 'helpers/rendering/render';
+import { trackComponentClick } from 'helpers/tracking/behaviour';
+import { GuardianTsAndCs } from 'pages/supporter-plus-landing/components/guardianTsAndCs';
+
+const darkBackgroundContainerMobile = css`
+	background-color: ${palette.neutral[97]};
+	${until.tablet} {
+		background-color: ${palette.brand[400]};
+		border-bottom: 1px solid ${palette.brand[600]};
+	}
+`;
+
+const columns = css`
+	position: relative;
+	color: ${palette.neutral[7]};
+	${textSans.medium()};
+	padding-top: ${space[2]}px;
+`;
+
+const secureTransactionIndicator = css`
+	margin-bottom: ${space[3]}px;
+	${from.tablet} {
+		margin-bottom: ${space[4]}px;
+	}
+`;
+
+const shorterBoxMargin = css`
+	:not(:last-child) {
+		${until.tablet} {
+			margin-bottom: ${space[2]}px;
+		}
+	}
+`;
+
+const searchParams = new URLSearchParams(window.location.search);
+const query = {
+	productId: searchParams.get('product'),
+	ratePlanId: searchParams.get('ratePlan'),
+	chargeId: searchParams.get('charge'),
+};
+
+const countryGroupIds = ['uk', 'us', 'eu', 'au', 'nz', 'ca', 'int'] as const;
+const CountryGroupIdSchema = picklist(countryGroupIds);
+const countryGroupId = parse(
+	CountryGroupIdSchema,
+	window.location.pathname.split('/')[1],
+);
+
+let currentCurrency: Currency;
+let currentCurrencyKey: keyof typeof currencies;
+switch (countryGroupId) {
+	case 'uk':
+		currentCurrency = currencies.GBP;
+		currentCurrencyKey = 'GBP';
+		break;
+	case 'us':
+		currentCurrency = currencies.USD;
+		currentCurrencyKey = 'USD';
+		break;
+	case 'au':
+		currentCurrency = currencies.AUD;
+		currentCurrencyKey = 'AUD';
+		break;
+	case 'eu':
+		currentCurrency = currencies.EUR;
+		currentCurrencyKey = 'EUR';
+		break;
+	case 'nz':
+		currentCurrency = currencies.NZD;
+		currentCurrencyKey = 'NZD';
+		break;
+	case 'ca':
+		currentCurrency = currencies.CAD;
+		currentCurrencyKey = 'CAD';
+		break;
+}
```

Comment:
I could have used our current transformers, but they are quite tied to redux. I am going to try stick to the 1 level deep `countryGroupId => T` transformations as that's all that's needed most the time.